packages/flutter/lib/src/painting/shader_warm_up.dart

```diff
@@ -178,5 +182,31 @@ class DefaultShaderWarmUp extends ShaderWarmUp {
     final ui.Paragraph paragraph = paragraphBuilder.build()
       ..layout(const ui.ParagraphConstraints(width: 60.0));
     canvas.drawParagraph(paragraph, const ui.Offset(20.0, 20.0));
+
+
+    // Construct an image for drawImage related operations
+    const int imageWidth = 40;
+    const int imageHeight = 40;
+    final Uint8List pixels = Uint8List.fromList(List<int>.generate(
+      imageWidth * imageHeight * 4,
+          (int i) => i % 4 < 2 ? 0x00 : 0xFF,  // opaque blue
+    ));
+
+    final Completer<void> completer = Completer<void>();
+    ui.decodeImageFromPixels(pixels, imageWidth, imageHeight, ui.PixelFormat.rgba8888, (ui.Image image) {
+      // Warm up image shaders
+      canvas.translate(0.0, 80.0);
+      canvas.save();
+      final ui.Rect srcRect = ui.Rect.fromLTWH(0.0, 0.0, image.width.toDouble(), image.height.toDouble());
+      canvas.drawImage(image, const ui.Offset(20.0, 20.0), ui.Paint());
+      canvas.translate(80.0, 0.0);
+      canvas.drawImageRect(image, srcRect, ui.Rect.fromLTWH(20.0, 20.0, 20.0, 20.0), paints[0]);
+      canvas.translate(80.0, 0.0);
+      canvas.drawImageRect(image, srcRect, ui.Rect.fromLTWH(10.0, 10.0, 60.0, 60.0), paints[0]);
+      canvas.restore();
+      completer.complete();
+    });
+
+    return completer.future;
```

Comment:
This PR had a substantial impact on our memory usage (increased by ~20%). This seems like a very high cost for an operation that will only benefit apps that have images.

Reply:
I'll decrease the image size from 40x40 to 10x10 ASAP which should decrease the memory usage 16x.

If the memory usage is still high after the change, I'll remove this part ASAP and put it in a custom shader warm up that's only used by a specific client.